### ISSUE TITLE
Intermittent test failure in PersistentTopicTest.testClosingReplicationProducerTwice

### ISSUE THREAD

**Issue by merlimat** [apache/incubator-pulsar#1145]
As seen in 

https://builds.apache.org/job/pulsar-pull-request/org.apache.pulsar$pulsar-broker/1344/testReport/junit/org.apache.pulsar.broker.service/PersistentTopicTest/testClosingReplicationProducerTwice/

```
org.mockito.exceptions.verification.TooManyActualInvocations: 

pulsarClientImpl.createProducerAsync(
    "persistent://prop/global/ns/testClosingReplicationProducerTwice",
    org.apache.pulsar.client.api.ProducerConfiguration@f74e835
);
Wanted 1 time:
-> at org.apache.pulsar.broker.service.PersistentTopicTest.testClosingReplicationProducerTwice(PersistentTopicTest.java:965)
But was 2 times. Undesired invocation:
-> at org.apache.pulsar.broker.service.AbstractReplicator.startProducer(AbstractReplicator.java:124)

	at org.apache.pulsar.broker.service.PersistentTopicTest.testClosingReplicationProducerTwice(PersistentTopicTest.java:965)

```

**Comment by merlimat** [comment]
This reproduces locally after running ~900 times.

**Comment by ivankelly** [comment]
This is a dupe of #617

**Comment by ivankelly** [comment]
There any logs beside that snippet?

**Comment by ivankelly** [comment]
Found the race. The builder stuff messed up this test completely, but the race is still there. Just put a 1s sleep after creating the replicator. Problem is ultimately that the mock should be applied before creating the replicator, because the replicator calls startProducer on construction, which will fail, and the call it again after a backoff, which is where the extra invokations are.

I'll submit a patch tomorrow.

**Comment by sijie** [comment]
This is fixed by #1360

**Comment by sijie** [comment]
The fix is contributed by @ivankelly
